connections/walletconnect.js

```diff
@@ -0,0 +1,128 @@
+const EventEmitter = require("events");
+const WC = require("@walletconnect/browser");
+const WCQRCode = require("@walletconnect/qrcode-modal");
+const dev = process.env.NODE_ENV === "development";
+
+let WalletConnect;
+
+class WalletConnectConnection extends EventEmitter {
+  constructor(_WalletConnect, url, options) {
+    super();
+
+    this.qrcode =
+      typeof options.qrcode === "undefined" || options.qrcode !== false;
+
+    WalletConnect = _WalletConnect || WC;
+
+    setTimeout(() => this.create(url, options), 0);
+  }
+
+  create(url, options) {
+    try {
+      this.wc = new WalletConnect({ bridge: url });
+    } catch (e) {
+      return this.emit("error", e);
+    }
+
+    if (!this.wc.connected) {
+      // create new session
+      this.wc
+        .createSession()
+        .then(() => {
+          if (this.qrcode) {
+            WCQRCode.open(this.wc.uri, () =>
+              this.emit(
+                "error",
+                new Error("User close WalletConnect QR Code modal")
```

Comment:
Typo. From

`new Error("User close WalletConnect QR Code modal")`

to 

`new Error("User closed WalletConnect QR Code modal")`

From `close` to `closed`

Reply:
👍

Reply:
👌